app/routes/surveys/components/SurveyEditor.js

```diff
@@ -3,42 +3,104 @@
 import styles from './surveys.css';
 import React, { Component } from 'react';
 import { DetailNavigation, ListNavigation } from '../utils.js';
+import Question from './Question';
 import { Field } from 'redux-form';
 import Button from 'app/components/Button';
 import {
   TextInput,
-  RadioButton,
+  CheckBox,
   SelectInput,
-  RadioButtonGroup,
   DatePicker
 } from 'app/components/Form';
 import { createValidator, required } from 'app/utils/validation';
 import { reduxForm } from 'redux-form';
 import type { SurveyEntity } from 'app/reducers/surveys';
 import { Content } from 'app/components/Content';
 import type { FieldProps } from 'redux-form';
+import Icon from 'app/components/Icon';
+import { Link } from 'react-router';
 
 type Props = FieldProps & {
   survey: SurveyEntity,
-  autoFocus: any,
+  autoFocus: Object,
   submitFunction: (SurveyEntity, ?number) => Promise<*>,
   deleteSurvey: number => Promise<*>,
   push: string => void
 };
 
-class SurveyEditor extends Component<Props> {
+type State = {
+  questions: Array<Object>
+};
+
+class SurveyEditor extends Component<Props, State> {
+  state = {
+    questions: []
+  };
+
+  componentDidMount() {
+    const questions =
+      this.props.survey.questions.length === 0
+        ? [
+            {
+              questionText: '',
+              questionType: 'single_choice',
```

Comment:
It would be nice to have all of these "constants" in one place, so you don't have to copy strings all the time while coding. Will help type-safety aswell